sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/ComplexQueriesTest.java

```diff
@@ -242,6 +244,185 @@ public void testComplex() throws IOException, InterruptedException {
         Utilities.compileAndTestRust(rustDirectory, true);
     }
 
+    // Test case for issue 740
+    @Test
```

Comment:
You probably want a more minimal test case too. I believe it's just "7 as seven" that triggers the bug right?

Reply:
I am not sure what does, didn't have time to reduce it.
But I don't see why that would have this effect.

Reply:
As per the issue, it's the only addition to the program that was otherwise compiling fine: https://github.com/feldera/feldera/issues/740

Reply:
yes, but I don't think it's in the inner select that the problem happens.
too tired to reduce the case now.